### PR TITLE
[pvr/cleanup/improvement] Some minor things I found while researching

### PR DESCRIPTION
**Pull request by Paxxi**
Observer crash in pvr

### CODE REVIEW

File: xbmc/epg/EpgContainer.cpp [hunk 1]
```diff
@@ -65,7 +65,10 @@ CEpgContainer::CEpgContainer(void) :
 
 CEpgContainer::~CEpgContainer(void)
 {
-  Unload();
+  // Don't call Unload here, it'll try
+  // to notify observers which may be in
+  // different states of teardown
+  Stop();
```
Comment (ksooo) [inline xbmc/epg/EpgContainer.cpp]: Imo, reaching dtor does not necessarily mean Kodi "shutdown". Thus I'm not sure this is okay.

Unload() does much more than Stop().

**Comment by ksooo** [comment]
Re: "Remove strange circular dependency way to trigger an epg update": good catch!

File: xbmc/epg/EpgContainer.cpp (hunk 1 above)
Comment (Paxxi) [inline xbmc/epg/EpgContainer.cpp]: Unload() does more stuff but when looking at it, none of it seemed important in a destructor as all objects will clean themselves up. I may be mistaken but it looked perfectly safe to remove the call to it.

File: xbmc/pvr/PVRManager.cpp [hunk 2]
```diff
@@ -365,6 +365,7 @@ void CPVRManager::SetState(ManagerState state)
     CSingleLock lock(m_managerStateMutex);
     m_managerState = state;
     SetChanged();
+    TriggerEpgsCreate();
```
Comment (ksooo) [inline xbmc/pvr/PVRManager.cpp]: Calling this on every (!) pvr manager state change looks horribly wrong to me (although in fact the existing code does eaxtly this). So, your change does not change behavior, but I doubt that the existing behavior is okay. We need to find a good condition for when to call TriggerEpgsCreate...

**Pull request closed without merging**